common.h

```diff
@@ -33,4 +33,6 @@ static inline uint64_t mmio_read64(void *addr)
 	return ((uint64_t) high << 32) | low;
 }
 
+#define fallthrough do {} while (0)
```

Comment:
All relative modern [compilers](https://gcc.gnu.org/onlinedocs/gcc/Statement-Attributes.html) know about fallthrough attribute/statement. Hence you should test if the compiler supports it.

The Linux kernel used to do something like this
```
#if __has_attribute(__fallthrough__)
# define fallthrough                    __attribute__((__fallthrough__))
#else
# define fallthrough                    do {} while (0)  /* fallthrough */
#endif
```
but IIRC it was now dropped as all supported gcc/clang versions know about fallthrough